spec/views/curriculum/units/_lessons_spec.rb

```diff
@@ -2,20 +2,41 @@
 
 RSpec.describe("curriculum/units/_lessons", type: :view) do
   let(:unit_json) { File.new("spec/support/curriculum/views/unit.json").read }
-
   let(:key_stage_json) { File.new("spec/support/curriculum/views/key_stage.json").read }
+  let(:user) { create(:user) }
+  let(:unit) {
+    JSON.parse(unit_json, object_class: OpenStruct).data.unit
+  }
+  let(:key_stage) {
+    JSON.parse(key_stage_json, object_class: OpenStruct).data.key_stage
+  }
 
-  before do
-    unit = JSON.parse(unit_json, object_class: OpenStruct).data.unit
-    key_stage = JSON.parse(key_stage_json, object_class: OpenStruct).data.key_stage
-    render partial: "lessons", locals: {key_stage: key_stage, unit: unit, lessons: unit.lessons}
-  end
+  context "user not logged in" do
+    before do
+      allow_any_instance_of(AuthenticationHelper).to receive(:current_user).and_return(nil)
+      render partial: "lessons", locals: {key_stage:, unit:, lessons: unit.lessons}
+    end
+    it "has a title" do
+      expect(rendered).to have_css(".govuk-heading-m", text: "Lessons")
+    end
 
-  it "has a title" do
-    expect(rendered).to have_css(".govuk-heading-m", text: "Lessons")
+    it "has list of lessons" do
+      expect(rendered).to have_css(".govuk-list", count: 1)
+    end
+
+    it "has login button" do
+      expect(rendered).to have_css(".govuk-button", text: "Log in to download")
+    end
   end
 
-  it "has list of lessons" do
-    expect(rendered).to have_css(".govuk-list", count: 1)
+  context "user logged in" do
+    before do
+      allow_any_instance_of(AuthenticationHelper).to receive(:current_user).and_return(user)
+      render partial: "lessons", locals: {key_stage:, unit:, lessons: unit.lessons}
+    end
+
+    it "has download button" do
+      expect(rendered).to have_css(".govuk-button", text: "Download all lesson files")
+    end
```

Comment:
We don't seem be testing the branch where there are no files to download. Would be good to make sure this test covers that case too unless we have a good reason not to.

Reply:
Totally agree, will add a test case for no files